### PR TITLE
Added converterFactories property to ApiClient in jvm-retrofit2.

### PR DESCRIPTION
**Pull request by dylan-kwon**
The existing `converterFactory` property worked fine, but was not available in certain cases.

I think there are these problems.
1. You can add only one `converterFactory`.
2. `converterFactory` can only be added at the very end.

In my case, I had to add my custom Converter.Factory first, because JsonParser doesn't handle empty strings and throws an exception. (#14982)

At first, I planned to add a `preConverterFactory` property, but changed `converterFactory` to an array to make it more customizable.

For backwards compatibility, `converterFactory` has not been removed from the code, it has been left using the @Deprecated annotation.

I think this PR of mine will help you meet more requirements.
 
### PR checklist

- [x] Read the [contribution guidelines](https://github.com/openapitools/openapi-generator/blob/master/CONTRIBUTING.md).
- [x] Pull Request title clearly describes the work in the pull request and Pull Request description provides details about how to validate the work. Missing information here may result in delayed response from the community.
- [x] Run the following to [build the project](https://github.com/OpenAPITools/openapi-generator#14---build-projects) and update samples:
  ```
  ./mvnw clean package 
  ./bin/generate-samples.sh
  ./bin/utils/export_docs_generators.sh
  ``` 
  Commit all changed files. 
  This is important, as CI jobs will verify _all_ generator outputs of your HEAD commit as it would merge with master. 
  These must match the expectations made by your contribution. 
  You may regenerate an individual generator by passing the relevant config(s) as an argument to the script, for example `./bin/generate-samples.sh bin/configs/java*`. 
  For Windows users, please run the script in [Git BASH](https://gitforwindows.org/).
- [x] In case you are adding a new generator, run the following additional script : 
  ```
  ./bin/utils/ensure-up-to-date.sh
  ``` 
  Commit all changed files.
- [x] File the PR against the [correct branch](https://github.com/OpenAPITools/openapi-generator/wiki/Git-Branches): `master` (6.3.0) (minor release - breaking changes with fallbacks), `7.0.x` (breaking changes without fallbacks)
- [x] If your PR is targeting a particular programming language, @mention the [technical committee](https://github.com/openapitools/openapi-generator/#62---openapi-generator-technical-committee) members, so they are more likely to review the pull request.

@jimschubert, @dr4ke616 @karismann @Zomzog @andrewemery @4brunu @yutaka0m @wing328

### CODE REVIEW

File: modules/openapi-generator/src/main/resources/kotlin-client/libraries/jvm-retrofit2/infrastructure/ApiClient.kt.mustache [hunk 1]
```diff
@@ -60,6 +60,28 @@ import okhttp3.MediaType.Companion.toMediaType
...
+        RxJava3CallAdapterFactory.create(),
+        {{/useRxJava3}}
+        {{#moshi}}
+        MoshiConverterFactory.create(serializerBuilder.build()),
+        {{/moshi}}
+        {{#kotlinx_serialization}}
+        kotlinxSerializationJson.asConverterFactory("application/json".toMediaType()),
+        {{/kotlinx_serialization}}
+    ),
+    @Deprecated("this property is deprecated. use converterFactories.")
```
Comment (4brunu) [inline modules/openapi-generator/src/main/resources/kotlin-client/libraries/jvm-retrofit2/infrastructure/ApiClient.kt.mustache:84]: Hey @dylan-kwon, thanks for creating this PR.
Could you please change the deprecation to something like this, that shows an error while trying to use converterFactory?
`@Deprecated("This property is deprecated. Please use converterFactories", replaceWith = ReplaceWith("converterFactories"), level = DeprecationLevel.ERROR)`
Thanks

**Comment by dylan-kwon** [comment]
Hi, @4brunu

`ci/circleci: node3 — Your tests failed on CircleCI` Any idea what could be causing this to fail?

**Review by 4brunu — APPROVED** [review]
Looks good to me

**Comment by 4brunu** [comment]
CI failure it's not related to this PR.

```
[ERROR] Failed to execute goal org.codehaus.mojo:exec-maven-plugin:1.2.1:exec (npm-test) on project TypeScriptBrowserPetstoreClientTests: Command execution failed.: Process exited with an error: 1 (Exit value: 1) -> [Help 1]
[ERROR] 
[ERROR] To see the full stack trace of the errors, re-run Maven with the -e switch.
[ERROR] Re-run Maven using the -X switch to enable full debug logging.
[ERROR] 
[ERROR] For more information about the errors and possible solutions, please read the following articles:
[ERROR] [Help 1] http://cwiki.apache.org/confluence/display/MAVEN/MojoExecutionException
[ERROR] 
[ERROR] After correcting the problems, you can resume the build with the command
[ERROR]   mvn <args> -rf :TypeScriptBrowserPetstoreClientTests
```

**Pull request merged**